worker/src/main/java/sbt/internal/worker1/ForkTestMain.java

```diff
@@ -325,6 +325,7 @@ private ExecutorService executorService(final boolean parallel) {
     }
 
     private void runTests(TestInfo info, ClassLoader classLoader) throws Exception {
+      Thread.currentThread().setContextClassLoader(classLoader);
```

Comment:
Nice work! Do you think we could test this via scripted test something like https://github.com/sbt/sbt/blob/develop/sbt-app/src/sbt-test/tests/fork2/build.sbt?